internal/shared/gen.go

```diff
@@ -0,0 +1,17 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shared // import "go.opentelemetry.io/otel/internal/shared"
+
+//go:generate go run ./gocpy -src=env.go -dest=../../exporters/jaeger/env_shared.go -pkg=jaeger
```

Comment:
Should this be the other way around?  Should this directive go in `exporters/jaeger/env_shared.go` to indicate how that file can be generated?  It's not clear how doing it this way will scale when more files and modules are involved.

Reply:
Totally agree. Addressed https://github.com/open-telemetry/opentelemetry-go/pull/3841/commits/9ede3f536e808365bcf0840a45be73e4141254bf